### PR TITLE
ddtrace/tracer: add IsTraceRoot to clients-side-stats

### PR DESCRIPTION
**Pull request by knusbaum**
### What does this PR do?

This PR adds the IsTraceRoot field to client stats payloads.

### Motivation

In order to support adaptive sampling, we want to make this small change to the existing client stats implementation to bring it up to date with the version in the Trace Agent.

See:
https://github.com/DataDog/datadog-agent/pull/23302
https://github.com/DataDog/datadog-agent/pull/25374

### Reviewer's Checklist

- [ ] Changed code has unit tests for its functionality at or near 100% coverage.
- [ ] [System-Tests](https://github.com/DataDog/system-tests/) covering this feature have been added and enabled with the va.b.c-dev version tag.
- [ ] There is a benchmark for any new code, or changes to existing code.
- [ ] If this interacts with the agent in a new way, a system test has been added.
- [ ] Add an appropriate team label so this PR gets put in the right place for the release notes.
- [ ] Non-trivial go.mod changes, e.g. adding new modules, are reviewed by @DataDog/dd-trace-go-guild.

Unsure? Have a question? Request a review!

### CODE REVIEW

File: ddtrace/tracer/stats.go [hunk 1]
```diff
@@ -278,13 +279,22 @@ func (sb *rawBucket) Export() statsBucket {
...
+)
+
 type rawGroupedStats struct {
 	hits            uint64
 	topLevelHits    uint64
 	errors          uint64
 	duration        uint64
 	okDistribution  *ddsketch.DDSketch
 	errDistribution *ddsketch.DDSketch
+	IsTraceRoot     trilean
```
Comment (darccio) [inline ddtrace/tracer/stats.go:297]: @knusbaum I didn't expect ternary logic 😆 What's the reason behind it? I assume it's avoiding possible nil pointers. Also, why an `int32`? Wouldn't an `int8` be enough?

**Review by darccio — APPROVED** [review]

File: ddtrace/tracer/stats.go (hunk 1 above)
Comment (darccio) [reply ddtrace/tracer/stats.go:297]: Ok, cool! 😁 Good to know.

**Review by darccio — APPROVED** [review]

File: ddtrace/tracer/stats.go [hunk 2]
```diff
@@ -278,13 +279,22 @@ func (sb *rawBucket) Export() statsBucket {
 	return csb
 }
 
+type trilean int32
+
+const (
+	trilean_not_set trilean = iota
+	trilean_true
+	trilean_false
```
Comment (felixge) [inline ddtrace/tracer/stats.go:287]: snake_case? 🙈

Comment (darccio) [reply ddtrace/tracer/stats.go:287]: @felixge You are right, I didn't notice it. The trilean concept blinded me. We'll merge to improve this.

**Pull request merged**